### DISCUSSION TITLE
What is HHAA's full name?

### DISCUSSION THREAD

**Discussion by caxieyou** [pixijs/pixijs#7991] [General]
Can not find it anywhere...

Thanks a lot!!!

**Comment by ivanpopelyshev** [comment]
Details on HHAA:

![image](https://user-images.githubusercontent.com/695831/141647308-093591aa-ad93-49f9-9756-28b50820635f.png) antialiasing

Its a way of moving from simple `clamp(x + 0.5, 0.0, 1.0)` to MITER , BEVEL and ROUND joins.

Video from khronos meetup: https://youtu.be/it6FkEzPUGY?t=823

Slides: https://www.khronos.org/developers/linkto/hww-to-draw-lines-in-webgl-crazy-panda

same slides in russian, but with bigger zoom and more detailed, page 47 and 49: https://assets.ctfassets.net/nn534z2fqr9f/6LM5n7g9ThY9VpRpq67AnT/ab6a8b31794bd59f9d8840e0f15d854d/______-________________-______-______________-__________.pdf

UPD.

Also , one guy from mapbox is making smooth lines for regl based on simple LAA: https://twitter.com/rickyreusser/status/1459289922796351491 , so from now on both PixiJS and regl have smooth lines.

LAA is also explained here a bit: https://acegikmo.com/shapes/docs/#anti-aliasing . Its just `clamp(x + 0.5, 0.0, 1.0)` with fade based on `fwidth(sdfLineParam)` (derivative)

**Reply by caxieyou** [reply]
thanks a hu ge
